### PR TITLE
Add sched_yield syscall.

### PR DESCRIPTION
**Pull request by j-piecuch**

### CODE REVIEW

File: include/sys/sched.h [hunk 1]
```diff
@@ -105,6 +105,12 @@ long sched_switch(void);
  */
 void sched_maybe_preempt(void);
 
+/*! \brief Unconditionally yield the CPU to another thread.
+ *
+ * \note Panics if interrupts or preemption are disabled.
+ */
+void sched_yield(void);
```
Comment (cahirwpz) [inline include/sys/sched.h:112]: `sched_yield` is defined in `include/sched.h` both on _Linux_ and _BSD_ systems.

File: sys/kern/sched.c [hunk 2]
```diff
@@ -211,6 +211,16 @@ void sched_maybe_preempt(void) {
   }
 }
 
+void sched_yield(void) {
```
Comment (cahirwpz) [inline sys/kern/sched.c:214]: In kernel space we can just call it `yield` as they do: http://bxr.su/NetBSD/sys/kern/kern_synch.c#272

File: include/sys/sched.h (hunk 1 above)
Comment (cahirwpz) [reply include/sys/sched.h:112]: I'm in favour of hiding most declarations from this file from user-land programs with simple trick we use in other header files: http://bxr.su/NetBSD/sys/sys/sched.h#143 (`#ifdef _KERNEL`)

Comment (j-piecuch) [reply include/sys/sched.h:112]: ...my bad, I forgot to add the `include/sched.h` file to the commit.

**Pull request merged**